service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -58,16 +60,25 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 		}
 	}
 
-	var clusterProfile clusterProfile
+	var determinedTCProfile clusterProfile
```

Comment:
I think this here deserves some unit tests.

Reply:
No unit test comes to mind which wouldn't be just re-implementation of the rules, so no added value. Logic is simple, very few self-explanatory rules. I've ensured it works as expected by testing cluster-operator together with nginx app on our test installations. Can you live with that for this iteration of essentially a PoC?

Reply:
Your responsibility. Your call.